### PR TITLE
Replace println! with assert! if possible in DataFusion examples

### LINKED ISSUE #11230: Replace `println!` with `assert!` if possible in DataFusion examples
we can use `assert!` in DataFusion examples to let the user know the expected output

_Originally posted by @comphead in https://github.com/apache/datafusion/pull/11088#discussion_r1662795223_

### PR DESCRIPTION
**Pull request by Nishi46**
Part of Closes #11230 

- Changed println statements to assert_eq in the relevant files.

### CODE REVIEW

**Comment by findepi** [comment]
please run `cargo fmt` to fix CI errors

**Review by alamb — APPROVED** [review]
Thank you @Nishi46 and @findepi  for the review  -- I kicked off the CI for this job plan to merge it if it passes

fyi @comphead

**Review by comphead — APPROVED** [review]
lgtm thanks @Nishi46 I think we need to fix 1 example to make it happen. 
Hope examples output is determenistic.

File: datafusion-examples/examples/sql_analysis.rs [hunk 1]
```diff
@@ -280,9 +280,11 @@ from
     // We can create a LogicalPlan from a SQL query like this
     let logical_plan = ctx.sql(tpcds_query_88).await?.into_optimized_plan()?;
 
-    println!(
-        "Optimized Logical Plan:\n\n{}\n",
-        logical_plan.display_indent()
+    assert_eq!(
+        logical_plan.display_indent().to_string(),
```
Comment (alamb) [inline datafusion-examples/examples/sql_analysis.rs:284]: this actual plan is quite a bit more complicated than shown here

**Review by alamb — COMMENTED** [review]
Thank you for this contribution @Nishi46  

The examples in rewrite_expr.rs are being split up (see https://github.com/apache/datafusion/pull/11085) so in order to help this PR along, I pushed some commits to the relevant tests.

Once this PR is merged I think subsequent PRs will be easer to get in because CI will run automatically for you

If you want to keep helping out here (and I hope that you do!) perhaps you can make multiple PRs, one for each example? I think that will also help get the code in sooner rather than later

**Pull request merged**